### PR TITLE
Interactive docs for models

### PR DESCRIPTION
**Pull request by smithis7**
**Description**

This adds more documentation for the TARDIS model, how a `Simulation` object is created, and a few other related things.

**How has this been tested?**
- [ ] Testing pipeline.
- [x] Other. 
Docs built locally and on github.

**Examples**

Main changes were done in this section: https://smithis7.github.io/tardis/branch/model_doc/physics/setup/index.html

**Type of change**

- [ ] Bug fix. 
- [ ] New feature. 
- [ ] Breaking change. 
- [x] None of the above. 

**Checklist**

- [x] My change requires a change to the documentation.
    - [x] I have updated the documentation accordingly.
    - [x] (optional) I have built the documentation on my fork following [the instructions](https://tardis-sn.github.io/tardis/development/documentation_preview.html).
- [x] I have assigned and requested two reviewers for this pull request.

### CODE REVIEW

File: docs/io/configuration/components/models/index.rst [hunk 1]
```diff
@@ -149,6 +149,8 @@ For example:
             density:
                 type: branch85_w7
 
+For more information, see `Branch85 W7 Density <../../../../physics/setup/model.ipynb#branch85-w7-density>`_.
```
Comment (wkerzendorf) [inline docs/io/configuration/components/models/index.rst:152]: can you absolute paths?

File: docs/physics/setup/model.ipynb [hunk 2]
```diff
@@ -0,0 +1,624 @@
+{
```
Comment (andrewfullard) [inline docs/physics/setup/model.ipynb:1]: <p>Typo/confusing sentence at "Using geometry..."</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/tardis-sn/tardis/pull/1849/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='23'/>

File: docs/io/configuration/components/models/index.rst (hunk 1 above)
Comment (smithis7) [reply docs/io/configuration/components/models/index.rst:152]: Unfortunately I cannot

File: docs/physics/setup/model.ipynb (hunk 2 above)
Comment (smithis7) [reply docs/physics/setup/model.ipynb:1]: Oops, I didn't finish that sentence

**Review by andrewfullard — APPROVED** [review]

File: docs/physics/setup/model.ipynb [hunk 3]
```diff
@@ -0,0 +1,624 @@
...
+   "source": [
+    "### Power Law Density\n",
+    "\n",
+    "The power law density profile uses the following equation:\n",
+    "\n",
+    "$$\\rho(v,t_0)=\\rho_0\\left(\\frac{v}{v_0}\\right)^{\\rm exponent}$$\n",
+    "and thus\n",
+    "$$\\rho(v,t_\\mathrm{explosion})=\\rho_0\\left(\\frac{v}{v_0}\\right)^{\\rm exponent}\\left(\\frac{t_0}{t_\\mathrm{explosion}}\\right)^3.$$\n",
+    "\n",
+    "Thus, the relevant parameters are $t_0$, $\\rho_0$, $v_0$, and the exponent. We reccomend that the exponent be negative for a sensible density profile. The schema can be found [here](../../io/configuration/components/models/index.rst#power-law-density).\n",
```
Comment (atharva-2001) [inline docs/physics/setup/model.ipynb:298]:
```suggestion
    "Thus, the relevant parameters are $t_0$, $\\rho_0$, $v_0$, and the exponent. We recommend that the exponent be negative for a sensible density profile. The schema can be found [here](../../io/configuration/components/models/index.rst#power-law-density).\n",
```

File: docs/physics/setup/model.ipynb [hunk 4]
```diff
@@ -0,0 +1,624 @@
...
+   "source": [
+    "## Abundance\n",
+    "\n",
+    "The `Radial1DModel` also carries important information about elemental abundances in each shell. These are mass abundances -- that is, the abundance of oxygen is the fraction of the shell's mass that is made up of oxygen.\n",
+    "\n",
+    "The only built-in abundance model that TARDIS offers is a uniform abundance, meaning each shell has identical abundances. Like density, however, the [file abundance](../../io/configuration/components/models/index.rst#file-abundance), [CSVY model](../../io/configuration/components/models/index.rst#csvy-model), and [custom abundance widget](../../io/visualization/abundance_widget.ipynb) methods allow users more freedom with assigning different abundances in each shell.\n",
+    "\n",
+    "A table of abundances in each shell is stored in the `abundance` attribute of the `Radial1DModel` object.\n",
+    "\n",
+    "In this first example, the abundances in each shell do not depend on the time after the explosion and are identical to those provides in the configuration."
```
Comment (atharva-2001) [inline docs/physics/setup/model.ipynb:403]:
```suggestion
    "In this first example, the abundances in each shell do not depend on the time after the explosion and are identical to those provided in the configuration."
```

File: docs/physics/setup/model.ipynb [hunk 5]
```diff
@@ -0,0 +1,624 @@
...
+    "\n",
+    "Next, TARDIS finds the temperature of the plasma in each cell, called the radiative temperature. In fact, this temperature changes throughout the simulation based on light-matter interations (see [Estimators and Convergence](../est_and_conv/index.ipynb)). When the model is initiated, TARDIS must calculate an initial guess for the radiative temperature in each shell.\n",
+    "\n",
+    "To do this, TARDIS also calculates the temperature at the surface of the photosphere $T_\\mathrm{inner}$. This is calculated as the temperature of a blackbody:\n",
+    "\n",
+    "$$T_{\\mathrm{inner}}=\\left(\\frac{L}{4 \\pi r_{\\mathrm{boundary\\_inner}}^2 \\sigma_{\\mathrm{R}}}\\right)^{1/4}$$\n",
+    "\n",
+    "where $\\sigma_{\\mathrm{R}}$ is the Stefan-Boltzmann constant, $r_{\\mathrm{boundary\\_inner}}$ is the radius of the photosphere, and $L$ is the luminosity of the photosphere (this is initially taken as the requested luminosity of the supernova though is changed throughout the simulation as part of the [convergence method](../est_and_conv/convergence.rst)). This calculation is also described in [Energy Packet Initialization](../montecarlo/initialization.ipynb).\n",
```
Comment (atharva-2001) [inline docs/physics/setup/model.ipynb:480]:
```suggestion
    "where $\\sigma_{\\mathrm{R}}$ is the Stefan-Boltzmann constant, $r_{\\mathrm{boundary\\_inner}}$ is the radius of the photosphere, and $L$ is the luminosity of the photosphere (this is initially taken as the requested luminosity of the supernova though it is changed throughout the simulation as part of the [convergence method](../est_and_conv/convergence.rst)). This calculation is also described in [Energy Packet Initialization](../montecarlo/initialization.ipynb).\n",
```

**Review by atharva-2001 — CHANGES_REQUESTED** [review]
Thank you for writing this documentation! I learnt a lot by reading it, but also found a bunch of typos-

**Comment by smithis7** [comment]
> Thank you for writing this documentation! I learnt a lot by reading it, but also found a bunch of typos-

Thank you so much. I am extremely bad at catching typos in my own writing.

File: docs/physics/setup/setup_example.ipynb [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+    "sim = Simulation.from_config(tardis_config)\n",
+    "sim"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "e3f86364",
+   "metadata": {},
+   "source": [
+    "When the `Simulation` object is created, a `Radial1DModel`, `BasePlasma`, and `MonteCarloRunner` object are created ans stored as attributes of the `Simulation` object."
```
Comment (atharva-2001) [inline docs/physics/setup/setup_example.ipynb:71]:
```suggestion
    "When the `Simulation` object is created, a `Radial1DModel`, `BasePlasma`, and `MonteCarloRunner` object are created and stored as attributes of the `Simulation` object."
```

File: docs/io/configuration/read_configuration.ipynb [hunk 7]
```diff
@@ -2,111 +2,147 @@
...
    "source": [
     "conf['montecarlo']['convergence_strategy']['damping_constant']"
-   ],
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Similarly to how they are accessed, entries in the configuration can be editted. For example,"
```
Comment (atharva-2001) [inline docs/io/configuration/read_configuration.ipynb:104]:
```suggestion
    "Similarly to how they are accessed, entries in the configuration can be edited. For example,"
```

**Review by atharva-2001 — CHANGES_REQUESTED** [review]
Thank you! Just a few more typo fixes and I think this PR should be ready to merge.

**Review by atharva-2001 — APPROVED** [review]

**Pull request merged**